src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,121 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int MAX_CAPACITY = 10;
+    private Object[] array = new Object[MAX_CAPACITY];
+    private int size;
+
+    public void grow() {
+        Object[] newArray = new Object[size + size / 2];
+        for (int i = 0; i < size; i++) {
+            newArray[i] = array[i];
+        }
+        array = newArray;
+    }
+
     @Override
     public void add(T value) {
-
+        array[size] = value;
+        size++;
+        if (size == array.length) {
+            grow();
+        }
```

Comment:
1. Please perform check before operation, not after. Probaly it was last insert by user. Why then we need to extend array?
2. you call grow() method only inside if block. Please include this check inside method
```suggestion
        growIfRequired();
```